src/result.zig

```diff
@@ -0,0 +1,148 @@
+const std = @import("std");
+const Statistics = @import("statistics.zig").Statistics;
+const format = @import("output/format.zig");
+const statistics = @import("statistics.zig");
+const Color = @import("output/color.zig").Color;
+const Runner = @import("runner.zig");
+const Readings = Runner.Readings;
+
+/// Carries the results of a benchmark. The benchmark name and the recorded
+/// durations are available, and some basic statistics are automatically
+/// calculated.
+pub const Result = struct {
+    name: []const u8,
+    readings: Readings,
+
+    pub fn init(name: []const u8, readings: Runner.Readings) !Result {
+        return Result{ .name = name, .readings = readings };
+    }
+
+    pub fn deinit(self: Result) void {
+        self.readings.deinit();
+    }
+
+    /// Formats and prints the benchmark result in a human readable format.
+    /// writer: Type that has the associated method print (for example std.io.getStdOut.writer())
+    /// colors: Whether to pretty-print with ANSI colors or not.
+    pub fn prettyPrint(
```

Comment:
since the Io.Writer gets rid of the anyerror returned from the old writer's anytype, we might as well use explicit error (Io.Writer.Error for instance). But this is probably better-placed in a separate PR.